emcc.py

```diff
@@ -2792,5 +2795,10 @@ def validate_arg_level(level_string, max_level, err_msg, clamp=False):
 
 
 if __name__ == '__main__':
-  run()
-  sys.exit(0)
+  try:
+    sys.exit(run())
+  except subprocess.CalledProcessError as e:
+    # Handle subprocess failure cleanly unles DEBUG is set
+    if DEBUG:
+      raise
+    sys.exit(e.returncode)
```

Comment:
in this case, would the error not even be shown?

Reply:
The backtrace would not be shown.

My rational is as follows:  In most cases the output from the failing tool is the most important information.  e.g. s2wasm tells you it failed for some reason.  In most cases we already catch this exception locally and print the command line that failed (see shared.py:execute and shared.py:check_execute).  So the user should see the output of the command and failed (i.e. the error message) and the command line itself that failed is also displayed.

What I don't really the user to have see in the normal use case is the python backtrace that led to that command.  Most users don't expect tooling to produce backtraces for user errors, and it can actually obscure the real issue.  For the llc case we already suppress this (via `exit_with_error('compiler frontend failed to generate LLVM bitcode, halting')`). 

With this change fewer users should be exposed to backtraces, which IMHO should be reserved to cases where the tool itself crashes, not on bad input.   The way I see it, a backtrace is like a segfault, if you see one you should file a bug in the tool.

Now that I think about this more though I think maybe the real fix should be in shared.py:execute  and shared.py:check_execute maybe they should exit on failure instead of re-raising the exception?

Reply:
Hmm, yeah, generally speaking that reasoning makes sense to me too.